javascript/modules/client/src/internal/interfaces/plugins.ts

```diff
@@ -13,91 +13,114 @@ import { BigNumber } from "@ethersproject/bignumber";
+  getProposals: (
+    params: IProposalQueryParams,
+  ) => Promise<AddressListProposalListItem[]>;
+  getSettings: (pluginAddress: string) => Promise<IPluginSettings | null>;
+}
+
+export interface IClientAddressListEncoding extends IClientCore {
+  updatePluginSettingsAction: (
+    pluginAddress: string,
+    params: IPluginSettings,
+  ) => DaoAction;
+  addMembersAction: (
+    pluginAddress: string,
+    members: string[],
+  ) => DaoAction;
+  removeMembersAction: (
+    pluginAddress: string,
+    members: string[],
+  ) => DaoAction;
+}
+export interface IClientAddressListDecoding extends IClientCore {
+  updatePluginSettingsAction: (data: Uint8Array) => IPluginSettings;
+  addMembersAction: (data: Uint8Array) => string[];
+  removeMembersAction: (data: Uint8Array) => string[];
+  findInterface: (data: Uint8Array) => IInterfaceParams | null;
+}
+export interface IClientAddressListEstimation extends IClientCore {
+  createProposal: (
+    params: ICreateProposalParams,
+  ) => Promise<GasFeeEstimation>;
+  voteProposal: (params: IVoteProposalParams) => Promise<GasFeeEstimation>;
+  executeProposal: (
+    params: IExecuteProposalParams,
+  ) => Promise<GasFeeEstimation>;
+}
 /** Defines the shape of the AddressList client class */
-export interface IClientAddressList extends IClientCore {
-  methods: {
-    createProposal: (
-      params: ICreateProposalParams,
-    ) => AsyncGenerator<ProposalCreationStepValue>;
-    voteProposal: (
-      params: IVoteProposalParams,
-    ) => AsyncGenerator<VoteProposalStepValue>;
-    executeProposal: (
-      params: IExecuteProposalParams,
-    ) => AsyncGenerator<ExecuteProposalStepValue>;
-    canVote: (params: ICanVoteParams) => Promise<boolean>;
-    getMembers: (addressOrEns: string) => Promise<string[]>;
-    getProposal: (propoosalId: string) => Promise<AddressListProposal | null>;
-    getProposals: (
-      params?: IProposalQueryParams,
-    ) => Promise<AddressListProposalListItem[]>;
-    getSettings: (pluginAddress: string) => Promise<IPluginSettings | null>;
-  };
-  encoding: {
-    /** Computes the parameters to be given when creating the DAO, so that the plugin is configured */
-    updatePluginSettingsAction: (
-      pluginAddress: string,
-      params: IPluginSettings,
-    ) => DaoAction;
-  };
-  decoding: {
-    updatePluginSettingsAction: (data: Uint8Array) => IPluginSettings;
-    findInterface: (data: Uint8Array) => IInterfaceParams | null;
-  };
-  estimation: {
-    createProposal: (
-      params: ICreateProposalParams,
-    ) => Promise<GasFeeEstimation>;
-    voteProposal: (params: IVoteProposalParams) => Promise<GasFeeEstimation>;
-    executeProposal: (
-      params: IExecuteProposalParams,
-    ) => Promise<GasFeeEstimation>;
-  };
+export interface IClientAddressList {
+  methods: IClientAddressListMethods;
+  encoding: IClientAddressListEncoding;
+  decoding: IClientAddressListDecoding;
+  estimation: IClientAddressListEstimation;
```

Comment:
I'm of the impression that we are typing the same thing several times. Maybe it's needed but we're typing these fields here and also typing them on the class itself.

Reply:
Yeah, we can skip typing them in the class, and it will infer the typing

Reply:
As you see fit, no personal preference